### ISSUE TITLE
`exclude` filters `.js` output, but not `.d.ts` (plugin `exclude` vs. `tsconfig` `exclude`)

### ISSUE THREAD

**Issue by sami616** [ezolenko/rollup-plugin-typescript2#225]
My **plugin config** :

```javascript
    typescript({
      typescript: require('typescript'),
      exclude: ['**/*.stories.tsx', 'node_modules'],
    }),
```

My include/exclude in **tsconfig**
```json
  "include": ["src/**/*", "typings/**/*"],
  "exclude": ["node_modules"]
```

Im seeing that the plugin is excluding the stories files from my output, however .d.ts files for them are stll being outputted:

Version:
```json
"rollup-plugin-typescript2": "^0.27.0",
```

**Comment by ltroller** [comment]
Can confirm same issue here.

**Comment by radufilipescu** [comment]
i am experiencing the same issue

**Comment by ajayvarghese** [comment]
I have this issue with `rollup-plugin-typescript2@0.27.1` `typescript@3.9.2`. But works after downgrading typescript to 3.8.3.

**Comment by agilgur5** [comment]
This looks like a duplicate of #226, which has the root cause. Per my comment there, encountered a similar issue (with stories too at that) in https://github.com/jaredpalmer/tsdx/pull/666#discussion_r404847759. That is due to `_.merge` doing a deep merge and a deep merge merges arrays based on _index_.

In OP's specific example, the fix is to flip the ordering so that `'node_modules'` is first and `'**/*.stories.tsx'` is second, i.e.

**plugin config** :
```js
  typescript({
    typescript: require('typescript'),
    exclude: ['node_modules', '**/*.stories.tsx'],
  }),
``` 
include/exclude in **tsconfig**

```json
  "include": ["src/**/*", "typings/**/*"],
  "exclude": ["node_modules"]
```

Or could just remove `"node_modules"` from the `tsconfig` `exclude` because that's included in [`tsconfig`'s defaults already](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#details)

**Comment by younho9** [comment]
Still same issue here..

**Comment by agilgur5** [comment]
This should be fixed by #347 

Once that's merged and released, the workaround of using separate `tsconfig` `exclude`s should no longer be necessary.

**Comment by agilgur5** [comment]
#347 has been released in [`0.33.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/0.33.0.1)
